core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -886,31 +884,39 @@ impl<R: WriteableRocks> StateManagerDatabase<R> {
     fn ensure_historical_substate_values(&self) {
         let db_context = self.open_rw_context();
         let extension_data_cf = db_context.cf(ExtensionsDataCf);
-        let values_associated_since = extension_data_cf
-            .get(&ExtensionsDataKey::ValuesAssociatedWithStateHashTreeSince)
-            .map(|bytes| scrypto_decode::<StateVersion>(&bytes).unwrap());
+        let status = extension_data_cf
+            .get(&ExtensionsDataKey::StateHashTreeAssociatedValuesStatus)
+            .map(|bytes| {
+                scrypto_decode::<VersionedStateHashTreeAssociatedValuesStatus>(&bytes)
+                    .unwrap()
+                    .into_latest()
+            });
 
         if self.config.enable_historical_substate_values {
-            if let Some(version) = values_associated_since {
-                info!("Historical Substate values enabled since {:?}", version);
+            if let Some(status) = status {
+                info!("Historical Substate values enabled since {:?}", status);
             } else {
                 let current_version = self.max_state_version();
                 info!(
                     "Enabling historical Substate values at {:?}",
                     current_version
                 );
                 self.populate_hash_tree_associated_substate_values(current_version);
+                let status = StateHashTreeAssociatedValuesStatusV1 {
+                    values_associated_from: current_version,
+                };
                 extension_data_cf.put(
-                    &ExtensionsDataKey::ValuesAssociatedWithStateHashTreeSince,
-                    &scrypto_encode(&current_version).unwrap(),
+                    &ExtensionsDataKey::StateHashTreeAssociatedValuesStatus,
+                    &scrypto_encode(&VersionedStateHashTreeAssociatedValuesStatus::from(status))
+                        .unwrap(),
                 );
             }
-        } else if let Some(version) = values_associated_since {
+        } else if let Some(status) = status {
             info!(
                 "Disabling historical Substate values (were enabled since {:?})",
-                version
+                status.values_associated_from
             );
-            extension_data_cf.delete(&ExtensionsDataKey::ValuesAssociatedWithStateHashTreeSince);
+            extension_data_cf.delete(&ExtensionsDataKey::StateHashTreeAssociatedValuesStatus);
```

Comment:
Shouldn't we delete all associated values here too? Otherwise it's a database leak I imagine?

Reply:
it's all a neverending database leak right now...

...luckily, there is a parallel PR with the GC impl, including the `delete_all()` call here :) https://github.com/radixdlt/babylon-node/pull/873/files#diff-5cf22ad2b534dd277f51718bd8cf5c3e624cecbc4819906008c39e0798427ce3R917